lib/matplotlib/artist.py

```diff
@@ -1593,52 +1597,59 @@ def getp(obj, property=None):
 
-    matplotlib supports the use of :func:`setp` ("set property") and
-    :func:`getp` to set and get object properties, as well as to do
-    introspection on the object.  For example, to set the linestyle of a
-    line to be dashed, you can do::
+    Parameters
+    ----------
+    obj : `.Artist` or list of `.Artist`
+        The artist(s) whose properties are being set or queried.  When setting
+        properties, all artists are affected; when querying the allowed values,
+        only the first instance in the sequence is queried.
 
-      >>> line, = plot([1, 2, 3])
-      >>> setp(line, linestyle='--')
+        For example, two lines can be made thicker and red with a single call:
 
-    If you want to know the valid types of arguments, you can provide
-    the name of the property you want to set without a value::
+        >>> x = arange(0, 1, 0.01)
+        >>> lines = plot(x, sin(2*pi*x), x, sin(4*pi*x))
+        >>> setp(lines, linewidth=2, color='r')
 
-      >>> setp(line, 'linestyle')
-          linestyle: {'-', '--', '-.', ':', '', (offset, on-off-seq), ...}
+    file : file-like or None, default: `sys.stdout`
+        Where `setp` writes its output, e.g. ::
+
+            with open('output.log') as file:
+                setp(line, file=file)
 
-    If you want to see all the properties that can be set, and their
-    possible values, you can do::
+    *args, **kwargs
+        The properties to set.  The following combinations are supported:
 
-      >>> setp(line)
-          ... long output listing omitted
+        - Set the linestyle of a line to be dashed:
 
-    By default `setp` prints to `sys.stdout`, but this can be modified using
-    the *file* keyword-only argument::
+          >>> line, = plot([1, 2, 3])
+          >>> setp(line, linestyle='--')
+
+        - Set multiple properties at once:
+
+          >>> setp(line, linewidth=2, color='r')
+
+        - List allowed values for a line's linestyle:
+
+          >>> setp(line, 'linestyle')
+          linestyle: {'-', '--', '-.', ':', '', (offset, on-off-seq), ...}
 
-      >>> with fopen('output.log') as f:
-      >>>     setp(line, file=f)
+        - List all properties that can be set, and their allowed values:
 
-    :func:`setp` operates on a single instance or a iterable of
-    instances. If you are in query mode introspecting the possible
-    values, only the first instance in the sequence is used. When
-    actually setting values, all the instances will be set.  e.g.,
-    suppose you have a list of two lines, the following will make both
-    lines thicker and red::
+          >>> setp(line)
+          agg_filter: a filter function, ...
+          [long output listing omitted]
 
-      >>> x = arange(0, 1, 0.01)
-      >>> y1 = sin(2*pi*x)
-      >>> y2 = sin(4*pi*x)
-      >>> lines = plot(x, y1, x, y2)
-      >>> setp(lines, linewidth=2, color='r')
+        `setp` also supports MATLAB style string/value pairs.  For example, the
```

Comment:
Single spaces after periods are currently winning:

```
$ find . -type f -name '*.py' | xargs grep '\. \S' | wc -l
2255
$ find . -type f -name '*.py' | xargs grep '\. \s' | wc -l
1396
```

and I don't see it anywhere in the style guide, so I assume the rich should get richer?

```suggestion
        `setp` also supports MATLAB style string/value pairs. For example, the
```

Reply:
I prefer double-spacing (actually I'd have to be quite careful to not double-space :)), and given that they're quite common in the codebase I'll stick to that unless we explicitly rule that we want single-spacing everywhere.